### DISCUSSION TITLE
Split Tunneling In Pc Client

### DISCUSSION THREAD

**Discussion by ZenMaxe** [vpnhood/VpnHood#257] [Q&A] [answered]
Hello How To Use Split Tunnel Feature in Pc Client?
it need config on server? how? i don't saw  any ip limiter in wiki

**Comment by epictz06** [comment]
Yes please

**Comment by trudyhood** [comment]
We have preconfigured the Spit Tunnelling on the client by excluding the users' client country.
For any other feature, you need to configure your own client:

For packet-level filtering, you can use the following option. We have a limitation in the number of masks
`you can use ClientOptions. PacketCaptureIncludeIpRanges`

For user-level filtering, you can use
`ClientOptions.IpFilter`

If you are using our App, you can also configure it at the app level:
'''
UserSettings.PacketCaptureIpRanges
UserSettings.PacketCaptureIpRangesFilterMode

UserSettings.IpGroupFilters
UserSettings.IpGroupFiltersMode
````

We have yet to provide any command to configure client IP splitting from the server.
I am sorry if it doesn't work as you expected.

**Reply by epictz06** [reply]
Thx for the reply ,
There is no way to put app id base tunneling? 
And what is the limitation

**Reply by ZenMaxe** [reply]
Oh Thank You 

I want provide a server only  for LoL Players (League of legends) (this means i want only tunnel LoL servers Not Others) for this job, iprange is not good but it's okay

**Reply by trudyhood** [reply]
I added this feature to our task list.

**Reply by trudyhood** [reply]
@ZenMaxe 
The feature is now available for use. To begin, navigate to Access Server > Server Profile and add the LoL server IPs. It is important to note that your host server should be static.  Once the IP addresses have been included, the client will only route the traffic that IPs. If you have any questions, please do not hesitate to ask.

**Reply by ZenMaxe** [reply]
Hello.
Wow Thank You for Adding This Feature

Can You Add a cli interface or database for this Feature?

I Want Add Ips With cli or editing file (i want write api Service for this)

**Reply by asoMansoury** [reply]
> Hello. Wow Thank You for Adding This Feature
> 
> Can You Add a cli interface or database for this Feature?
> 
> I Want Add Ips With cli or editing file (i want write api Service for this)

Hi ZenMaxe.
Can you explain by scenario about this feature that you have submitted.
I didn't understand what was your requirement and what you were looking for.
thank you.

**Reply by ZenMaxe** [reply]
Simple example:
- i want create a asp net web api in server (which have vpnhood) and connect that to app or telegram bot.
Than i add ip/routes in telegram or app
Than my web service add them in server (for split Tunneling)

**Reply by asoMansoury** [reply]
Can I have your telegram ID?
this is my email.
is it possible to send me your telegram id?
aso.mansoury@gmail.com
thank you.

**Reply by epictz06** [reply]
> @ZenMaxe The feature is now available for use. To begin, navigate to Access Server > Server Profile and add the LoL server IPs. It is important to note that your host server should be static. Once the IP addresses have been included, the client will only route the traffic that IPs. If you have any questions, please do not hesitate to ask.

Is it imposibble to spilit tunnel like android? Choosing app we want and done?

**Reply by ZenMaxe** [reply]
> Can I have your telegram ID? this is my email. is it possible to send me your telegram id? [aso.mansoury@gmail.com](mailto:aso.mansoury@gmail.com) thank you.

I sent you the ID

**Comment by trudyhood** [comment, marked as answer]
@ZenMaxe
We have REST API and .NET library to manage access server programmatically.
https://api.vpnhood.com/swagger/index.html
https://www.nuget.org/packages/VpnHood.AccessServer.Api

**Reply by ZenMaxe** [reply]
Thanks, Where is The Ip Section? ( Section that can Add Route for Split Tunneling)

**Reply by trudyhood** [reply]
> Thanks, Where is The Ip Section? ( Section that can Add Route for Split Tunneling)

If you are using VpnHood access server, go to Server Farms, and select the Server Profile. 
Use "Kernel Include IP Addresses"  OR "Include IP Addresses"

![image](https://user-images.githubusercontent.com/72891124/226999700-6ffd6df0-8d7f-46f7-9a26-34904c8e5c14.png)

**Reply by ZenMaxe** [reply]
Yeah i saw this

I mean any api is exist for this section? (Ip include)

**Reply by trudyhood** [reply]
> Yeah i saw this
> 
> I mean any api is exist for this section? (Ip include)

Everything that you can do in the VpnHood Access Server can be done by API. Check the swagger I sent to you. 
https://api.vpnhood.com/swagger/index.html#/ServerProfiles/ServerProfiles_Update

serverConfig parameter is a json file. To see the JSON format click the advanced button:
![image](https://user-images.githubusercontent.com/72891124/227012712-30925753-1227-4c8b-a2e6-c268c6a34a8f.png)

this may be helpful too:
https://github.com/vpnhood/VpnHood/wiki/VpnHood-Server-Configuration

**Reply by ZenMaxe** [reply]
Thanks i found it

**Comment by ZenMaxe** [comment]
hello @trudyhood 

can we do whitelist by domain list or process name?
can we have this feature?

**Comment by trudyhood** [comment]
* Filter by DomainName is already implemented. However, it has not been released and not tested properly. It is also not reliable because it is almost impossible to detect the domains of new protocols such as QUIC. We still have some ideas to achieve that on the client, but it is not on our priority list.

* Filter by process-name is not on our plan yet
